### PR TITLE
[12.0] [MIG] Product Brand

### PR DESCRIPTION
**Pull request by ThomasBinsfeld**
Migration of product_attribute from 11.0 to 12.0.
#389

### CODE REVIEW

**Review by acsonefho — APPROVED** [review]
LGTM (about code)

**Review by mpanarin — APPROVED** [review]

File: product_brand/models/product_brand.py
Comment (SimoRubi) [inline product_brand/models/product_brand.py]: Please substitute `©` with `Copyright` to follow the format defined in *.py files of https://github.com/OCA/maintainer-tools/blob/master/template/module

File: product_brand/readme/USAGE.rst
Comment (SimoRubi) [inline product_brand/readme/USAGE.rst:14]: Grouping by brand results in:
```
Odoo Server Error
Traceback (most recent call last):
  File "/.repo_requirements/odoo/odoo/http.py", line 653, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/.repo_requirements/odoo/odoo/http.py", line 312, in _handle_exception
    raise pycompat.reraise(type(exception), exception, sys.exc_info()[2])
  File "/.repo_requirements/odoo/odoo/tools/pycompat.py", line 87, in reraise
    raise value
  File "/.repo_requirements/odoo/odoo/http.py", line 695, in dispatch
    result = self._call_function(**self.params)
  File "/.repo_requirements/odoo/odoo/http.py", line 344, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/.repo_requirements/odoo/odoo/service/model.py", line 97, in wrapper
    return f(dbname, *args, **kwargs)
  File "/.repo_requirements/odoo/odoo/http.py", line 337, in checked_call
    result = self.endpoint(*a, **kw)
  File "/.repo_requirements/odoo/odoo/http.py", line 938, in __call__
    return self.method(*args, **kw)
  File "/.repo_requirements/odoo/odoo/http.py", line 517, in response_wrap
    response = f(*args, **kw)
  File "/home/odoo/OCB-12.0/addons/web/controllers/main.py", line 955, in call_kw
    return self._call_kw(model, method, args, kwargs)
  File "/home/odoo/OCB-12.0/addons/web/controllers/main.py", line 947, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/.repo_requirements/odoo/odoo/api.py", line 745, in call_kw
    return _call_kw_model(method, model, args, kwargs)
  File "/.repo_requirements/odoo/odoo/api.py", line 718, in _call_kw_model
    result = method(recs, *args, **kwargs)
  File "/.repo_requirements/odoo/odoo/models.py", line 2062, in read_group
    result = self._read_group_raw(domain, fields, groupby, offset=offset, limit=limit, orderby=orderby, lazy=lazy)
  File "/.repo_requirements/odoo/odoo/models.py", line 2175, in _read_group_raw
    self._cr.execute(query, where_clause_params)
  File "/.repo_requirements/odoo/odoo/sql_db.py", line 148, in wrapper
    return f(self, *args, **kwargs)
  File "/.repo_requirements/odoo/odoo/sql_db.py", line 225, in execute
    res = self._obj.execute(query, params)
psycopg2.ProgrammingError: column sale_report.product_brand_id does not exist
LINE 3: ...uct_brand" as "sale_report__product_brand_id" ON ("sale_repo...
```

**Review by SimoRubi — CHANGES_REQUESTED** [review]
code review and tested on runbot

File: product_brand/__manifest__.py
Comment (nikul-serpentcs) [inline product_brand/__manifest__.py:23]: @ThomasBinsfeld Improve Sequence as per [this](https://github.com/odoo/odoo/blob/11.0/addons/hr_recruitment/__manifest__.py#L20)

**Review by nikul-serpentcs — COMMENTED** [review]
Small Change

File: product_brand/__manifest__.py
Comment (ThomasBinsfeld) [reply product_brand/__manifest__.py:23]: @nikul-serpentcs I'm not sure to understand.

Comment (nikul-serpentcs) [reply product_brand/__manifest__.py:23]: @ThomasBinsfeld  IMO improve file position like [this](https://github.com/OCA/maintainer-tools/blob/master/template/module/__manifest__.py#L27).

always first `security` and after `views` and `reports`.

**Comment by pedrobaeza** [comment]
Please check Travis and runbot status

**Comment by pedrobaeza** [comment]
This also misses commits from #365

**Comment by pedrobaeza** [comment]
@ThomasBinsfeld can you please review #364 and incorporate it here if everything is correct?

**Comment by pedrobaeza** [comment]
#364 has been merged

File: product_brand/__manifest__.py
Comment (ThomasBinsfeld) [reply product_brand/__manifest__.py:23]: Done

File: product_brand/readme/USAGE.rst
Comment (ThomasBinsfeld) [reply product_brand/readme/USAGE.rst:14]: Should be solved now

**Comment by ThomasBinsfeld** [comment]
@pedrobaeza All 11.0 commits are here now

**Comment by pedrobaeza** [comment]
2 little pylint things:
```
product_brand/models/product_brand.py:31: [C8108(method-compute), ProductBrand] Name of compute method should start with "_compute_"
product_brand/reports/sale_report.py:15: [W0102(dangerous-default-value), SaleReport._query] Dangerous default value {} as argument
```

**Comment by pedrobaeza** [comment]
@SimoRubi @nikul-serpentcs can you review again?

File: product_brand/reports/account_invoice_report.py
Comment (bealdav) [inline product_brand/reports/account_invoice_report.py:20]: can be summarized by

`        return super()._select() + ", sub.product_brand_id as product_brand_id"
`

Just typo, not blocking

The same below

**Comment by SimoRubi** [comment]
@ThomasBinsfeld please check runbot and Travis, everything is red

File: product_brand/__manifest__.py
Comment (lmignon) [inline product_brand/__manifest__.py:13]: @ThomasBinsfeld Can you set the `development_status`key?

Comment (ThomasBinsfeld) [reply product_brand/__manifest__.py:13]: Done

**Review by angelmoya — APPROVED** [review]
:+1:

File: product_brand/tests/test_product_brand.py
Comment (nikul-serpentcs) [inline product_brand/tests/test_product_brand.py:12]: @ThomasBinsfeld Here `self.env.ref` not `self.ref`

**Review by nikul-serpentcs — CHANGES_REQUESTED** [review]
Minor Change

File: product_brand/tests/test_product_brand.py
Comment (ThomasBinsfeld) [reply product_brand/tests/test_product_brand.py:12]: @nikul-serpentcs `self.ref` is a good way to get the id of the record and `self.env.ref` a good way to get the record itself. I won't change it since we only need the ID in this unit test. Please feel free to submit a refactoring PR after the migration merge.

File: product_brand/reports/account_invoice_report.py
Comment (ThomasBinsfeld) [reply product_brand/reports/account_invoice_report.py:20]: @bealdav Thanks for your review. I don't think we should refactor code in migration PR if it's not required. Feel free to submit a refactoring in a PR after the migration merge.

File: product_brand/tests/test_product_brand.py
Comment (nikul-serpentcs) [reply product_brand/tests/test_product_brand.py:12]: @ThomasBinsfeld In case, In future use supplier record then change the `self.env.ref`.
meanwhile, We change this(self.env.ref) and using `self.supplier.id`.

It's not a blocking issue but  It's Good for us.

rather than, Code review LGTM.

Comment (ThomasBinsfeld) [reply product_brand/tests/test_product_brand.py:12]: @ThomasBinsfeld If you're review is non-blocking, you can update your review and add an approval. Thanks for the review.

**Review by nikul-serpentcs — APPROVED** [review]
Code Revie LGTM :+1:

File: product_brand/reports/account_invoice_report.py
Comment (bealdav) [reply product_brand/reports/account_invoice_report.py:20]: OK. No problem

**Comment by youring** [comment]
Can pick https://github.com/OCA/product-attribute/pull/114, or add as an option?

**Comment by ThomasBinsfeld** [comment]
@youring I don't plan to forward port  improvements from previous versions in this migration. But feel free to open a pull request targeting my branch and I will review/merge it.

**Comment by ghost** [comment]
Could this PR merged?

**Comment by pedrobaeza** [comment]
Please squash migration commits together.

**Comment by ThomasBinsfeld** [comment]
@pedrobaeza Done

**Pull request merged**
